### ISSUE TITLE
[FEAT]: Use System Folder as RAG's Knowledge Base

### ISSUE THREAD

**Issue by shgsousa** [Mintplex-Labs/anything-llm#1425]
### What would you like to see?

I’m utilizing Anything LLM as a knowledge assistant to discuss business processes and share my own insights, which I periodically record in documents. However, I’m facing a challenge keeping the system up-to-date with the evolving nature of these files, as they are not static documents.

My ideal solution would be a RAG system that enables me to:

1. Directly access answers from a set of files within a designated folder (and possibly subfolders) on my computer. To avoid scanning, splitting, and embedding unnecessary files in case of subfolders, I’d like the ability to specify file extension filters.
2. Automatically detect changes to the documents and rebuild the vector database at a scheduled interval, during agent use, or when deemed necessary.
3. Ensure that the vector database update occurs in memory, in the background, without disrupting agent usage that can continue working with outdated content until the new context data becomes available. Once the new embeddings and indexing are complete. This would enable Anything LLM to reduce the time of locking the system for usage only to the time it takes to serialize changes to the vector database, and loading the new vector database to memory for the operation of the agent.

Implementing this feature would significantly enhance the suitability of Anything LLM for my intended use case. Additionally, I’m impressed with the project’s progress and intend to contribute financially from time to time.

**Comment by lctuck** [comment]
This is right on target.  Adding, tracking, updating files is a real issue.  Pointing to a folder(s)/subfolder(s) would absolutely be a next level enhancement for this wonderful piece of tech.

**Comment by timothycarambat** [comment]
If anyone is watching this, live file sync for remote files (web, Github, etc) is already "live" in beta https://docs.useanything.com/beta-preview/active-features/live-document-sync and the soon to be merged branch [desktop-local-file-sync](https://github.com/Mintplex-Labs/anything-llm/tree/desktop-local-file-sync) will expand that to any locally upload file for desktop only.

Permissions disable us from being able to do this with local files for self-hosting and obviously remote hosted instances. So local file sync will only ever be on desktop

**Comment by tino** [comment]
The mentioned branch doesn't exist anymore, however I don't yet see this in v1.6.4? Or am I missing something?

**Comment by timothycarambat** [comment]
It is in the app under experimental:
https://docs.anythingllm.com/beta-preview/enable-feature

Then you can enable it. It is under there because without a doubt someone will try to proactively track 10K files and wonder why it uses so many resources & that is currently out of scope for its current support.

https://docs.anythingllm.com/beta-preview/active-features/live-document-sync
